overlord/fdestate/fdemgr.go

```diff
@@ -35,23 +40,46 @@ var (
 
 // FDEManager is responsible for managing full disk encryption keys.
 type FDEManager struct {
-	state *state.State
+	state   *state.State
+	initErr error
+
+	preseed bool
+	mode    string
 }
 
 type fdeMgrKey struct{}
 
-func Manager(st *state.State, runner *state.TaskRunner) *FDEManager {
+func initModeFromModeenv(m *FDEManager) error {
+	mode, explicit, err := boot.SystemMode("")
+	if err != nil {
+		return err
+	}
+
+	if explicit {
```

Comment:
we probably want a comment here that we are happy for mode to be "" on older systems as anyway then the FDE manager is not relevant

Reply:
added